src/js/stores/BallotStore.js

```diff
@@ -110,14 +110,20 @@ class BallotStore extends FluxMapStore {
     let newBallot = {};
 
     switch (action.type) {
+      // Dale 2016-06-02 On some pages we are calling BallotActions.retrieve() as many as three times in order to
+      // avoid problems like Issue #224. This should be optimized.
 
       case "voterRetrieve":
+        // console.log("BallotStore, voterRetrieve");
+        // We need to refresh the ballot here (in addition to voterAddressRetrieve) because there are routes that you
+        // can go to directly (e.g. wevote.me/ballot) that don't reach out and retrieve the ballot items any other way.
         BallotActions.retrieve();
```

Comment:
Hmm, did you try the code without this? I don't think it should matter which route you're on. voterRetrieve is called by the application, then voterAddressRetrieve is called, and ballotItems shouldn't be retrieved twice.